Remove the requirement to run as superuser

We'll set the appropriate capabilities to the completed binary itself,
so it can run without route at runtime (and also doesn't need to set
them itself with the PRIVDROP feature).

--- a/main.c	2021-12-16 12:17:42.000000000 +0000
+++ b/main.c	2022-08-18 10:10:30.654053602 +0100
@@ -550,9 +550,6 @@
     }
   }
 
-  if (user_check && getuid() != 0)
-    LOG_FATAL("Not superuser");
-
   /* Turn into a daemon */
   if (!nofork) {
     go_daemon();
